File: 이재진/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    ls = list(map(int, list(str(storey))))
    tmp = 0
    for i in range(len(ls)-1,-1,-1):
        now = ls[i] + tmp
        if now > 5 or (now == 5 and i > 0 and ls[i-1] >= 5):
            answer += 10-now
            tmp = 1
        else:
            answer += now
            tmp = 0
    if tmp == 1:
        answer += tmp
    return answer

File: 이재진/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int answer = 0;
        while (storey != 0) {
            int x = storey%10;
            if (x == 0) {
                storey = (int) storey/10;
            } else if (x == 5) {
                if (storey % 100 == 0 || storey % 100 >= 50) {
                    storey += 10-x;
                    answer += 10-x;
                } else {
                    storey -= x;
                    answer += x;
                }
            } else if (x > 5) {
                storey += 10-x;
                answer += 10-x;
            } else {
                storey -= x;
                answer += x;
            }
            
        }
        return answer;
    }
}
